### PR TITLE
Add unit test

### PR DESCRIPTION
**Pull request by penjc**
# Description

Add unit test. All are generated by gpt.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by penjc** [comment]
i don't know how to fix this.
    "Copyright (c) 2018 Oleksii Trekhleb\n",
    "Copyright (c) 2020 Samuel Huang\n",
    "Copyright (c) 2023 jerry-git\n",
but it shows up as one line in the book.

**Comment by zhangqi444** [comment]
> i don't know how to fix this. "Copyright (c) 2018 Oleksii Trekhleb\n", "Copyright (c) 2020 Samuel Huang\n", "Copyright (c) 2023 jerry-git\n", but it shows up as one line in the book.

Just give an empty lines in between each of them

File: assignments/prerequisites/python-programming-advanced.ipynb [hunk 1]
```diff
@@ -45,6 +45,23 @@
     "# Python programming advanced"
```
Comment (zhangqi444) [inline assignments/prerequisites/python-programming-advanced.ipynb:45]: <pre class="ql-syntax" spellcheck="false">Line #24.        def test_raises_exception(self):
</pre><p><code>test_wrong_target_type()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/346/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Comment by zhangqi444** [comment]
All code is good. How you come up with such test cases? If you use GPT, please call it out in the PR description.

**Pull request merged**